openff/toolkit/_tests/test_molecule.py

```diff
@@ -3695,6 +3424,281 @@ def test_deepcopy_not_shallow(self):
+        assert qcschema.mass_numbers.tolist() == qca_mol.mass_numbers.tolist()
+        assert qcschema.name == qca_mol.name
+        assert qcschema.masses.all() == qca_mol.masses.all()
+        assert qcschema.molecular_charge == qca_mol.molecular_charge
+        assert qcschema.molecular_multiplicity == qca_mol.molecular_multiplicity
+        assert qcschema.real.all() == qca_mol.real.all()
+
+    def test_qcschema_round_trip_from_to_from(self):
+        """Test making a molecule from qca record using from_qcschema,
+        then converting back to qcschema using to_qcschema,
+         and then reading that again using from_qcschema"""
+
+        # get a molecule qcschema
+        import qcportal
+
+        client = qcportal.PortalClient("https://api.qcarchive.molssi.org:443")
+
+        ds = client.get_dataset(
+            "torsiondrive", "OpenFF-benchmark-ligand-fragments-v1.0"
+        )
+        # grab an entry from the torsiondrive data set
+        entry = ds.get_entry(
+            "[H]c1[c:1]([c:2](c(c(c1[H])N([H])C(=O)[H])[H])[C:3]2=C(C(=C([S:4]2)[H])OC([H])([H])[H])Br)[H]".lower()
+        )
+        # now make the molecule from the record instance with the geometry
+        mol_qca_record = Molecule.from_qcschema(entry, client)
+        off_qcschema = mol_qca_record.to_qcschema()
+        mol_using_from_off_qcschema = Molecule.from_qcschema(off_qcschema)
+        assert_molecule_is_equal(
+            mol_qca_record,
+            mol_using_from_off_qcschema,
+            "Molecule roundtrip to/from_qcschema failed",
+        )
+
+    def test_qcschema_round_trip_raise_error(self):
+        """Test making a molecule from qcschema,
+        reaching inner except block where everything fails"""
+        import qcportal
+
+        client = qcportal.PortalClient("https://api.qcarchive.molssi.org:443")
+
+        ds = client.get_dataset(
+            "torsiondrive", "OpenFF-benchmark-ligand-fragments-v1.0"
+        )
+        # grab an entry from the torsiondrive data set
+        entry = ds.get_entry(
+            "[H]c1[c:1]([c:2](c(c(c1[H])N([H])C(=O)[H])[H])[C:3]2=C(C(=C([S:4]2)[H])OC([H])([H])[H])Br)[H]".lower()
+        )
+        del entry.attributes["canonical_isomeric_explicit_hydrogen_mapped_smiles"]
+        # now make the molecule from the record instance with the geometry
+        with pytest.raises(KeyError):
+            Molecule.from_qcschema(entry, client)
+
+    @pytest.mark.flaky(reruns=10)
+    def test_qcschema_molecule_record_round_trip_from_to_from(self):
+        """Test making a molecule from qca record using from_qcschema,
+        then converting back to qcschema using to_qcschema,
+         and then reading that again using from_qcschema"""
+
+        # get a molecule qcschema
+        import qcportal
+
+        client = qcportal.PortalClient("https://api.qcarchive.molssi.org:443")
+
+        record = [*client.query_molecules(molecular_formula="C16H20N3O5")][-1]
```

Comment:
This was previously `[0]`, which regularly grabbed a record that did not have CMILES. There's probably a different index that's safer, since this assumes that all future records will have CMILES that the underlying toolkits can assign stereochemistry to, but that assumption seems relatively safe.